service/status-report/main.go

```diff
@@ -77,6 +85,43 @@ func main() {
 		klog.Exitf("Failed to get storageClient %q/%q: %v", storageClient.Namespace, storageClient.Name, err)
 	}
 
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err = cl.List(ctx, &csvList, client.InNamespace(storageClientNamespace)); err != nil {
+		klog.Exitf("Failed to list csv resources: %v", err)
+	}
+
+	var opVersion string
+	for idx := range csvList.Items {
+		candidate := &csvList.Items[idx]
+		if strings.HasPrefix(candidate.Name, csvPrefix) {
+			opVersion = candidate.Spec.Version.String()
+			break
+		}
+	}
+
+	if opVersion == "" {
+		klog.Exitf("Unable to find csv with prefix %q", csvPrefix)
+	}
+
+	clusterVersion := &configv1.ClusterVersion{}
+	clusterVersion.Name = "version"
+	if err = cl.Get(ctx, types.NamespacedName{Name: clusterVersion.Name}, clusterVersion); err != nil {
+		klog.Exitf("Failed to get clusterVersion: %v", err)
+	}
+
+	var cVersion string
+	for idx := range clusterVersion.Status.History {
+		candidate := &clusterVersion.Status.History[idx]
+		if candidate.State == configv1.CompletedUpdate {
+			cVersion = candidate.Version
+			break
+		}
+	}
```

Comment:
With the utils function in place, this can be simplified into
```suggestion
	record := utils.find(clusterVersion.Status.History, func (record *<type>) bool {
		return record.state == configv1.CompletedUpdate
	})
```

Reply:
done.